bundle-requests.js

```diff
@@ -0,0 +1,85 @@
+import { sendRequest } from "./send-request.js";
+import { createHash } from "node:crypto";
+
+// map from bundle key to bundle object
+const bundles = {};
+
+const getBundleKey = (entry, muSessionId, changeSets) => {
+  if (entry.options.preciseBundling) {
+    // more precise regarding allowed groups BUT slower since we're hashing a possibly longish string
+    const hash = createHash("sha256");
```

Comment:
since it doesn't really matter here, why not use the computational less expensive sha1?

Reply:
wanted less risk of collision, but if you all agree that this is fine, this will indeed make it faster

Reply:
Why not just add mu-auth-allowed-groups to the end of the hash-key?  We know it's a JSON encoded string so we know it will not collide with the index number.  We can let the hashing function take care of it disambiguating and it will correctly take care of unlikely collisions.

That would speed it up by which we could make it the default.

The bigger oversight here would be that the order of the hash-keys in the array is not relevant and that the order of the keys is neither.  I don't think we need to tackle that for this PR either.  It catches the vast majority of the cases.

Reply:
indeed the order is a problem in theory. In practice it seems to be always the same though. i'll add the json encoded string to the key and remove the precise bundling option